src/test/java/htsjdk/samtools/SAMSequenceRecordTest.java

```diff
@@ -39,4 +42,42 @@ public void testGetSAMString() {
         r.setMd5("7a6dd3d307de916b477e7bf304ac22bc");
         Assert.assertEquals("@SQ\tSN:chr5_but_without_a_prefix\tLN:271828\tSP:Psephophorus terrypratchetti\tAS:GRCt01\tM5:7a6dd3d307de916b477e7bf304ac22bc", r.getSAMString());
     }
+
+    @DataProvider
+    public Object[][] testIsSameSequenceyData() {
+        final SAMSequenceRecord rec1 = new SAMSequenceRecord("chr1", 100);
+        final SAMSequenceRecord rec2 = new SAMSequenceRecord("chr2", 101);
+        final SAMSequenceRecord rec3 = new SAMSequenceRecord("chr3", 0);
+        final SAMSequenceRecord rec4 = new SAMSequenceRecord("chr1", 100);
+
+        final String md5One = "1";
+        final String md5Two = "2";
+        final int index1 = 1;
+        final int index2 = 2;
+
+        return new Object[][]{
+                new Object[]{rec1, rec1, md5One, md5One, index1, index1, true},
+                new Object[]{rec1, null, md5One, md5One, index1, index1, false},
+                new Object[]{rec1, rec4, md5One, md5One, index1, index1, true},
+                new Object[]{rec1, rec4, md5One, md5One, index1, index2, false},
+                new Object[]{rec1, rec3, md5One, md5Two, index1, index1, false},
+                new Object[]{rec1, rec2, md5One, md5Two, index1, index1, false},
+                new Object[]{rec1, rec4, md5One, null, index1, index1, true},
+        };
+    }
+
+    @Test(dataProvider = "testIsSameSequenceyData")
+    public void testIsSameSequence(final SAMSequenceRecord rec1 , final SAMSequenceRecord rec2, final String md5One, final String md5Two,
+                                   final int index1, final int index2, final boolean isSame) {
+        if (rec2 != null) {
```

Comment:
Already checking that rec2 is null, can't do rec1 is null since it will only be testing the test because `rec1.isSameSequence(rec2)` and will throw an NPE.  Added a test for identical records except for their sequence index.